HOLD - Documentation version 1 - Limited pilot with prototyping partners

anikbrazeau opened the pull request:
# Description

First version of the documentation site is deploying from `docs` branch. This PR would bring it into the `main` branch so that it's _more_ in the open.
https://cds-snc.github.io/forms-api/ 

# Implications

If this PR gets merged, it will require updating the "Settings" for the repo to deploy the "main" branch instead of "docs". There is some remaining work to be done to rework the architecture to modernize it -- a few things in the config file are not even needed.

![image](https://github.com/user-attachments/assets/1dac37ee-581a-42a1-858b-7cfbfde54005)

# Reference links
https://docs.google.com/document/d/1HbTmNL8rSnYl_GnCT0hACPL6CNq1j6UOiR8-w0DqWgQ/edit?tab=t.0
https://github.com/cds-snc/pspc-documentation/tree/master
https://github.com/cds-snc/talent
https://github.com/cds-snc/rcmp-report-a-cybercrime-alpha-documentation

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update menu.yml

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update index.html

anikbrazeau committed:
Update accueil.html

anikbrazeau committed:
Update menu.yml

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Create authentification.md

anikbrazeau committed:
Update authentification.md

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Create pour-commencer.md

anikbrazeau committed:
Create surveiller.md

anikbrazeau committed:
Create obtenir-du-soutien.md

anikbrazeau committed:
Create faire-des-requetes.md

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update index.html

anikbrazeau committed:
Update index.html

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Delete docs/_includes/home.md

anikbrazeau committed:
Delete docs/_includes/accueil.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Update obtenir-du-soutien.md

anikbrazeau committed:
Update surveiller.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Create home.md

anikbrazeau committed:
Create accueil.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Create retrieving-new-form-submissions.md

anikbrazeau committed:
Create retrieving-specific-form-submissions.md

anikbrazeau committed:
Create confirming-form-submissions

anikbrazeau committed:
Rename confirming-form-submissions to confirming-form-submissions.md

anikbrazeau committed:
Create reporting-a-problem-with-a-form-submission

anikbrazeau committed:
Create getting-form-questions.md

anikbrazeau committed:
Create submenu.html

anikbrazeau committed:
Update sidebar.html

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update menu.yml

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update sidebar.html

anikbrazeau committed:
Update sidebar.html

anikbrazeau committed:
Update sidebar.html

anikbrazeau committed:
Update submenu.html

anikbrazeau committed:
Update menu.yml

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update default.html

anikbrazeau committed:
Delete docs/_pages/en/confirming-form-submissions.md

anikbrazeau committed:
Delete docs/_pages/en/retrieving-new-form-submissions.md

anikbrazeau committed:
Delete docs/_pages/en/retrieving-specific-form-submissions.md

anikbrazeau committed:
Delete docs/_pages/en/reporting-a-problem-with-a-form-submission

anikbrazeau committed:
Delete docs/_pages/en/getting-form-questions.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update getting-support.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update Gemfile

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Create style.scss

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update home.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update authentification.md

anikbrazeau committed:
Update faire-des-requetes.md

anikbrazeau committed:
Update obtenir-du-soutien.md

anikbrazeau committed:
Update pour-commencer.md

anikbrazeau committed:
Update surveiller.md

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update accueil.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update getting-support.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update authentication.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update getting-support.md

craigzour committed:
Merge remote-tracking branch 'origin/main' into docs

craigzour committed:
biome should ignore docs folder

anikbrazeau committed:
Update accueil.md

FR for home

anikbrazeau committed:
Update pour-commencer.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update pour-commencer.md

anikbrazeau committed:
Update authentification.md

FR docs for auth

anikbrazeau committed:
Update faire-des-requetes.md

Change title FR

anikbrazeau committed:
Update menu.yml

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update and rename faire-des-requetes.md to effectuer-des-demandes.md

FR for requests

anikbrazeau committed:
Update surveiller.md

anikbrazeau committed:
Update obtenir-du-soutien.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update pour-commencer.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update Gemfile.lock

Attempt to update vulnerable dependencies

anikbrazeau committed:
Update Gemfile.lock

Fixing vulnerabilities based on outdated versions

anikbrazeau committed:
Update Gemfile.lock

anikbrazeau committed:
Update Gemfile.lock

anikbrazeau committed:
Update Gemfile.lock

anikbrazeau committed:
Update Gemfile.lock

anikbrazeau committed:
Update Gemfile.lock

craigzour commented:
@anikbrazeau I had a discussion with Bryan about the PR. Before merging to our main branch we should find some time to do some clean-up in the different files that compose the documentation. I saw things that I am pretty sure are not needed (configuration lines, dependencies, etc..).

This is something we could tackle when we start our tech debt sprint. For now let's keep the documentation in that `docs` branch.

anikbrazeau committed:
Add throttling / rate limiting to v1 docs (#105)

* Update monitoring.md

Add API rate limit / throttling content

* Update monitoring.md

* Update monitoring.md

* Update surveiller.md

* Update monitoring.md

* Update monitoring.md

* Update monitoring.md

* Update surveiller.md

* Update surveiller.md

* Update monitoring.md

* Update surveiller.md

* Update monitoring.md

* Update monitoring.md

* Update surveiller.md

* Update monitoring.md

* Update surveiller.md

* Update surveiller.md

* Update monitoring.md

* Update docs/_pages/en/monitoring.md

* Update docs/_pages/fr/surveiller.md

* Update docs/_pages/en/monitoring.md

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update normalize.css

anikbrazeau committed:
Update syntax.css

anikbrazeau committed:
Update normalize.css

anikbrazeau committed:
Update normalize.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update normalize.css

anikbrazeau committed:
Update syntax.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update normalize.css

anikbrazeau committed:
Update _config.yml

anikbrazeau committed:
Update main.css

anikbrazeau committed:
Update monitoring.md

anikbrazeau committed:
Update surveiller.md

anikbrazeau committed:
Update getting-started.md

anikbrazeau committed:
Update pour-commencer.md

anikbrazeau committed:
[ON HOLD] v2 API Docs: Self-serve selecting API delivery method within GC Forms and creating/deleting API key  (#112)

* Update getting-started.md

* Update pour-commencer.md

* Update home.md

* Update accueil.md

* Update home.md

* Update translations.yml

* Update monitoring.md

* Update surveiller.md

* Update getting-started.md

* Update pour-commencer.md

* Update getting-started.md

anikbrazeau committed:
chore: update API docs --> Protected B  (#161)

* Update subtitle to Protected B

* Add PB to getting started page EN

* Add PB to getting started page FR

* Update docs/_pages/fr/pour-commencer.md

Co-authored-by: Clément JANIN <ninjaclem8@hotmail.fr>

---------

Co-authored-by: Clément JANIN <ninjaclem8@hotmail.fr>

anikbrazeau committed:
chore: Documentation update for question IDs and group tags (#171)

* Update making-requests.md

Typo

* Update making-requests.md

English doc update

* Update effectuer-des-demandes.md

French updates

* Update making-requests.md

Match French

* Update docs/_pages/fr/effectuer-des-demandes.md

* Apply suggestions from code review

Update what you can do with tags

* Update docs/_pages/en/making-requests.md

Co-authored-by: Clément JANIN <ninjaclem8@hotmail.fr>

---------

Co-authored-by: Clément JANIN <ninjaclem8@hotmail.fr>

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update making-requests.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update effectuer-des-demandes.md

anikbrazeau committed:
Update translations.yml

anikbrazeau committed:
Update version number

anikbrazeau committed:
chore: Update references to Settings in API docs (#246)

* Modify homepage tag line in translations

Updated the homepage tag line in both English and French translations to remove 'Protected B' as everything is now protected B, no need to differentiate.

* API key refresh EN

* Revise API key refresh to key rotation FR

Updated the section on API key management to clarify the process of key rotation and its necessity.

* Update API key instructions in getting-started.md

Clarified API key generation instructions and security note.

* Add API key requirement to documentation

anikbrazeau committed:
Update version number in translations.yml

anikbrazeau committed:
chore: Update documentation for file attachment feature  (#238)

* Modify homepage tag line in translations

Updated the homepage tag line in both English and French translations to remove 'Protected B' as everything is now protected B, no need to differentiate.

* API key refresh EN

* Revise API key refresh to key rotation FR

Updated the section on API key management to clarify the process of key rotation and its necessity.

* Update API key instructions in getting-started.md

Clarified API key generation instructions and security note.

* Add API key requirement to documentation

* Document file retrieval process for submissions - EN

Added section on retrieving attached files from submissions, including security considerations and example response data.

* Add details on retrieving form submission attachments - FR

Added section on retrieving attachments from form submissions, including security considerations and example response data.

* Revise documentation on form submission file retrieval

Updated section headers and clarified file retrieval process.

* Enhance security section for file retrieval

Clarified the process of retrieving attached files securely, emphasizing the importance of checking the isPotentiallyMalicious attribute for safety.

* Enhance security instructions for file analysis

Added guidance on handling the 'isPotentiallyMalicious' attribute for security.

* Update warning about file analysis responsibility

Clarified responsibility for checking the 'isPotentiallyMalicious' attribute before opening linked files.

* Clarify file scanning responsibility in documentation

Clarified responsibility for checking the isPotentiallyMalicious attribute in file scanning.

* Update file retrieval explanation in documentation

Clarified that files are retrieved from AWS and updated the wording for better understanding.

* Update explanation for retrieving attached files

Clarified that files are retrieved from AWS and linked to responses.

* Update file retrieval instructions with guidance link

Added a link to the Guidance page for file uploads in form submissions.

* Update API file retrieval instructions

Added a reference link to the guide for uploading files in the API section.

* Revise JSON example for file attachments

Updated the JSON structure for file attachments in the documentation, adding fields for 'id', 'downloadLink', and 'md5'.

* Revise JSON example for file attachments

Updated the JSON structure for attached files in the documentation.

* File scanning warning

* Apply suggestion from @anikbrazeau

* Apply suggestion from @anikbrazeau

anikbrazeau committed:
Update Base URL for authentication endpoint in docs (#257)

* Update Base URL for authentication endpoint

* Change base URL to include versioning

Updated the base URL for the API documentation.

anikbrazeau committed:
Correct link for file retrieval guidance

Fixed a broken link in the documentation regarding file retrieval and updated the guidance page link.

anikbrazeau committed:
Update file retrieval instructions in French documentation

anikbrazeau committed:
Fix link formatting in making-requests.md

anikbrazeau committed:
Add note about timezone based on beta user feedback (#271)

* Add note about timezone based on beta user feedback

Added a note about the timestamp being in UTC.

* Update notes in FR for timestamp

anikbrazeau committed:
Update homepage tag line version to 1.7.1
